src/main/client_context.cpp

```diff
@@ -350,6 +351,41 @@ ClientContext::CreatePreparedStatement(ClientContextLock &lock, const string &qu
 	return result;
 }
 
+shared_ptr<PreparedStatementData>
+ClientContext::CreatePreparedStatement(ClientContextLock &lock, const string &query, unique_ptr<SQLStatement> statement,
+                                       optional_ptr<case_insensitive_map_t<Value>> values) {
+	// check if any client context state could request a rebind
+	bool can_request_rebind = false;
+	for (auto const &s : registered_state) {
+		if (s.second->CanRequestRebind()) {
+			can_request_rebind = true;
+			break;
+		}
+	}
+	if (can_request_rebind) {
+		// if any registered state can request a rebind we do the binding on a copy first
+		try {
+			return CreatePreparedStatementInternal(lock, query, statement->Copy(), values);
+		} catch (std::exception &ex) {
+			ErrorData error(ex);
+			// check if any registered client context state wants to try a rebind
+			bool rebind = false;
+			for (auto const &s : registered_state) {
+				auto info = s.second->OnPlanningError(*this, *statement, error);
+				if (info == RebindQueryInfo::ATTEMPT_TO_REBIND) {
+					rebind = true;
+					break;
```

Comment:
I would like to suggest to remove this break statement. Our OnPlanningError() will open a distributed transaction, but if this would be a query between a postgres and motherduck table, now it could happen that the postgres OnPlanningError is called first, answeres ATTEMPT_REBIND, and then motherduck's OnPlanningError() is not called. So please invoke all callbacks.

Reply:
Sure, I've removed the break